### PR TITLE
Force resolution of sub dependency `coa` to prevent security issues

### PR DESCRIPTION
**Pull request by jtwillis92**
### Summary of Changes
The `coa` npm package was recently hijacked to introduce malicious code in versions > 2.0.2. ([reference](https://www.bleepingcomputer.com/news/security/popular-coa-npm-library-hijacked-to-steal-user-passwords/)). We are currently running the correct version per our `yarn.lock` file but to prevent the possibility of unintentional upgrades to the infected version we should enforce all dependencies to resolve `coa` to 2.0.2 by use of the `resolutions` tag in package.json.

### How to Test
The only test for this is running `yarn install` and then ensuring that the `coa` version is at 2.0.2 as expected.

> *Demo GIF(s) and screenshots for testing procedure*

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

+ [x] Look up the acceptance criteria in the related issue; paste ACs below in checklist format.
+ [x] Check against the criteria:
   + [x] prevent upgrades to `coa`  sub-dependency by forcing dependencies that rely on it to always resolve to the latest, safe version
  + [x] reviewed and tested by @ADPennington 

As Product Owner, @lfrohlich will decide if ACs are met.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)
  + [x] Backend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [ ] Does this PR change any linting or CI settings?

### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*
+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with @iamjolly and @ttran-hub using Accessibility Insights reveal any errors introduced in this PR?
    + [See the full Accessibility Assessment plan here.](https://github.com/HHS/TANF-app/blob/main/docs/Technical-Documentation/how-government-will-test-a11y.md)
_n/a_

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful deployment by assigning a 'Deploy with CircleCI - {env}' label*

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

+ [ ] If this PR introduces backend code, is that code documented both inline and overall?
+ [x] If this PR introduces frontend code, is that code documented both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [x] Does the OWASP Scan pass on CircleCI?
+ [x] Do manual code review and manual testing detect any security issues?
_no new issues detected_

### Deliverable 8: Context

> * Performance Standard(s): Code must be understandable and contextualized for the reviewers possess the knowledge and background necessary for analysis and constructive criticism to take place.

> * Acceptable Quality Level: Code submitted in the pull request has context.*
+ [x] Does this pull request contain sufficient inline comments providing relevant context for code snippets?
+ [x] Is the code understandable and lucid?
+ [x] Does this pull request provide background for why coding decisions were made?
+ [x] Can you as a reviewer explain and take ownership of these elements presented in this code review?

### CODE REVIEW

**Review by abottoms-coder — APPROVED** [review]

File: tdrs-frontend/package.json [hunk 1]
```diff
@@ -28,8 +28,9 @@
     "uuid": "^8.3.2"
   },
   "resolutions": {
-    "lodash": "^4.17.20",
     "bl": "^4.0.3",
+    "coa": "2.0.2",
```
Comment (jorgegonzalez) [inline tdrs-frontend/package.json:32]: Nit, but when we add resolutions like this, we also need to run `yarn` to get an updated `yarn.lock`

Comment (jtwillis92) [reply tdrs-frontend/package.json:32]: It actually results in the same yarn.lock:
```
❯ yarn                                                                                                                                                                                                                                                                ─╯
yarn install v1.22.15
[1/4] Resolving packages...
warning Resolution field "ssri@8.0.1" is incompatible with requested version "ssri@^6.0.1"
success Already up-to-date.
Done in 1.09s.
```

**Comment by valcollignon** [comment]
@abottoms-coder @jtwillis92 and @ADPennington to pair on this outside of dev sync.
After this is merged, does anything need to change in our process?

**Comment by jtwillis92** [comment]
> @abottoms-coder @jtwillis92 and @ADPennington to pair on this outside of dev sync. After this is merged, does anything need to change in our process?

No, this doesn't functionally change anything actually - it just prevents us from running into unintentional upgrades to this package.

**Review by ADPennington — APPROVED** [review]
thanks for the quick action on this @jtwillis92 🚀 whenever we do the dependency audit again (#838), probably worth checking in on `coa` again to see if the issue has been addressed so we aren't unnecessarily continuing to resolve to this version. @abottoms-coder ready to merge.

**Pull request merged**